### PR TITLE
Fixed editor log path and improved editor log not found error string

### PR DESCRIPTION
**Pull request by AMZN-AlexOteiza**

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]
I am assuming this change is to fix not finding the Editor.log on Linux.
Especially since the branch is linux_automation

**Review by Kadino — APPROVED** [review]
Valid fix for Linux, should also be a valid change for Mac. Change to Windows is technically unnecessary but keeps it in line with the others.

**Comment by AMZN-AlexOteiza** [comment]
> Valid fix for Linux, should also be a valid change for Mac. Change to Windows is technically unnecessary but keeps it in line with the others.

It works in Windows because of casing, but sometimes it can even cause problems.

The code was wrong. It should be "Editor" and not "editor".

**Comment by lumberyard-employee-dm** [comment]
> 
> 
> > Valid fix for Linux, should also be a valid change for Mac. Change to Windows is technically unnecessary but keeps it in line with the others.
> 
> It works in Windows because it ignores casing, but sometimes it can even cause problems.
> 
> The code was wrong. It should be "Editor" and not "editor".

That is the reason the `pathlib.PurePath` __eq__ operator should be used for path comparisons and not string comparisons

File: Tools/LyTestTools/tests/unit/test_editor_test_utils.py [hunk 1]
```diff
@@ -119,15 +119,15 @@ def test_RetrieveEditorLogContent_CrashLogExists_ReturnsLogInfo(self, mock_retri
         mock_log = 'mock log info'
 
         with mock.patch('builtins.open', mock.mock_open(read_data=mock_log)) as mock_file:
-            assert f'[editor.log]  {mock_log}' == editor_test_utils.retrieve_editor_log_content(0, mock_logname, mock_workspace)
+            assert f'[{mock_logname}]  {mock_log}' == editor_test_utils.retrieve_editor_log_content(0, mock_logname, mock_workspace)
```
Comment (evanchia-ly-sdets) [inline Tools/LyTestTools/tests/unit/test_editor_test_utils.py:122]: I would prefer if this assert would be:
`assert mock_log in editor_test_utils.retrieve_editor_log_content(0, mock_logname, mock_workspace)`

**Pull request merged**